lib/spack/llnl/util/filesystem.py

```diff
@@ -472,3 +474,110 @@ def find_library_path(libname, *paths):
         if os.path.exists(library):
             return path
     return None
+
+# Utilities for libraries
+
+
+class LibraryList(collections.Sequence):
+    """Manages a list of absolute paths to libraries"""
+    def __init__(self, libraries):
+        self.libraries = list(libraries)
+
+    @property
+    def directories(self):
+        return [os.path.dirname(x) for x in self.libraries]
+
+    @property
+    def libnames(self):
+        return [os.path.basename(x).split('.')[0] for x in self.libraries]
+
+    @property
+    def names(self):
```

Comment:
matter of convention, but maybe call this one libname as (from [here](http://tldp.org/HOWTO/Program-Library-HOWTO/shared-libraries.html) )

> The soname has the prefix lib, the name of the library, the phrase .so, followed by a period and a version number that is incremented whenever the interface changes (as a special exception, the lowest-level C libraries don't start with lib). 

in other words `libFOO.so` has a library name `FOO`.

One would need to figure out how to call the property above then.